src/components/chat/commented-documents.tsx

```diff
@@ -36,109 +25,30 @@ export const CommentedDocuments: React.FC<IProps> = ({user, handleDocView}) => {
 
-  //------Curriculum Documents: (i.e. //"Problem"/"Teacher-Guide")
-  const [docsCommentedOn, setDocsCommentedOn] = useState<PromisedCurriculumDocument[]>();
-  const cDocsRef = useMemo(() => db.collection("curriculum"), [db]);
-  const cDocsInScopeRef = useMemo(() => {
-  if (user?.network){
-    return  cDocsRef
-    .where("unit", "==", unit)
-    .where("problem", "==", problem)
-    .where("network","==", user?.network);
-  } else {
-    return  cDocsRef
-    .where("unit", "==", unit)
-    .where("problem", "==", problem)
-    //for teachers not in network, look for documents matching the uid
-    .where ("uid", "==", user?.id);
-  }
-  }, [cDocsRef, problem, unit, user?.network, user?.id]);
-
-  //------Curriculum Documents: (i.e. //"Problem"/"Teacher-Guide")
-  useEffect(() => {
-    const unsubscribeFromDocs = cDocsInScopeRef.onSnapshot(querySnapshot => {
-      const docs = querySnapshot.docs.map(doc => {
-        return (
-          {
-            id: doc.id,
-            title: "temp",
-            numComments: 0,
-            ...doc.data()
-          }
-        );
-      });
-      const commentedDocs: PromisedCurriculumDocument[] = [];
-      const promiseArr: Promise<void>[] = [];
-      for (let doc of docs){
-        const docCommentsRef = cDocsRef.doc(doc.id).collection("comments");
-        promiseArr.push(docCommentsRef.get().then((qs) => {
-          if (qs.empty === false) {
-            const firstCharPosition = doc.id.split("_", 4).join("_").length + 1; //first char after 4th _
-            const sectionType =  doc.id.substring(firstCharPosition, doc.id.length);
-            doc = {...doc, title: getSectionTitle(sectionType), numComments: qs.size};
-            commentedDocs.push(doc as PromisedCurriculumDocument);
-          }
-        }));
-      }
-      Promise.all(promiseArr).then((results)=>{
-        setDocsCommentedOn(commentedDocs);
-      });
-    });
-    return () => unsubscribeFromDocs?.();
-  },[cDocsRef, cDocsInScopeRef]);
+  const [commentedDocumentsQuery, setCommentedDocumentsQuery] = useState<CommentedDocumentsQuery>();
+  const [curricumDocs, setCurricumDocs] = useState<CurriculumDocumentInfo[]>([]);
+  const [studentDocs, setStudentDocs] = useState<StudentDocumentInfo[]>([]);
 
 
-  //------Documents: (i.e. //"Student Workspaces/"My Work"/"Class Work")
-  const [workDocuments, setWorkDocuments] = useState<PromisedDocumentDocument[]>();
-  const mDocsRef = useMemo(() => db.collection("documents"), [db]);
-  const mDocsInScopeRef = useMemo(() => {
-    if(user?.network){
-      return mDocsRef.where("network", "==", user?.network);
-    } else {
-      return mDocsRef.where("uid", "==", user?.id);
+  useEffect(() => {
+    if (user) {
+      setCommentedDocumentsQuery(new CommentedDocumentsQuery(db, user, unit, problem));
     }
-  }, [mDocsRef, user?.network, user?.id]);
+  }, [user, db, unit, problem]);
 
-  //------Documents: (i.e. //"Student Workspaces/"My Work"/"Class Work")
   useEffect(() => {
```

Comment:
Changed.  A couple of curiosity questions:
- Why `setUser` in an effect rather than going one step further and doing everything in the `useState` callback?
- "when the queries update them, it should trigger a component re-render" -- how does this happen?